sdk/provisioning/Azure.Provisioning/src/BicepListOfT.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
+
+namespace Azure.Provisioning;
+
+/// <summary>
+/// Represents a list of Bicep values.
+/// </summary>
+/// <typeparam name="T">The type of element represented by the values.</typeparam>
+public class BicepList<T> : BicepValue, IList<BicepValue<T>>
```

Comment:
`BicepList<T>` and `BicepDictionary<T>` provide strongly typed collections.  Dictionaries currently require `string` keys because I haven't seen anything in ARM that doesn't use `string` or something equivalent like a `ResourceIdentifier`.  We'll want to flip on the rest of the RPs before we GA things to make sure there are no surprises.  You can assign expressions to either the entire collection or individual elements.